pkg/kv/testdriver/store.go

```diff
@@ -0,0 +1,112 @@
+package testdriver
```

Comment:
I think you reimplemented the code of mockgen, it gives you exactly what you need here.